pkg/table/tables/tables.go

```diff
@@ -1211,7 +1211,7 @@ func (t *TableCommon) RemoveRecord(ctx table.MutateContext, h kv.Handle, r []typ
 	}
 
 	// The table has non-public column and this column is doing the operation of "modify/change column".
-	if len(t.Columns) > len(r) && t.Columns[len(r)].ChangeStateInfo != nil {
+	if !ctx.HasExtraInfo() && len(t.Columns) > len(r) && t.Columns[len(r)].ChangeStateInfo != nil {
```

Comment:
Look like `ctx.HasExtraInfo()` will always be true in `DELETE` DML, Does this affect the logic of the original if judgment?

Reply:
Hahaha... insert on duplicate update will do the deletion. And this if branch is just designed for that case...

Insert on duplicate will use the whole columns(the same as deletable columns). It just uses writable columns.

Reply:
My confusion is why we skip to handle the non-public column in normal deletion. Because the missing column data has already been processed outside `RemoveRecord` ? If so, could we move the below codes totally outside, it is really confusing here...